test/test_fusion_profiler.cpp

```diff
@@ -21,7 +21,52 @@
 
 namespace nvfuser {
 
-class FusionProfilerTest : public NVFuserTest {};
+class FusionProfilerTest : public NVFuserTest {
+ protected:
+  void SetUp() override {
+    NVFuserTest::SetUp();
+    if (ProfilerOptionsGuard::getCurOptions().hasAny()) {
+      const auto& current_profiling_opts =
+          ProfilerOptionsGuard::getCurOptions();
+      for (ProfOptsUnderType opt =
+               static_cast<ProfOptsUnderType>(ProfilerOption::Enable);
+           static_cast<ProfilerOption>(opt) != ProfilerOption::EndOfOption;
+           ++opt) {
+        const auto profiling_opt = static_cast<ProfilerOption>(opt);
+        if (current_profiling_opts.has(profiling_opt)) {
+          stored_data_[profiling_opt] =
+              current_profiling_opts.getArgs(profiling_opt);
+        }
+      }
+    }
```

Comment:
Can we simply store `ProfilerOptionsGuard::getCurOptions()` into an `Options<ProfilerOption>`? Why do we need to run a for loop to individually store the value of each option?

Reply:
The main is that there is no access to data placeholder within `Options<ProfilerOption>` - we could have a copy of the object but then, when we restore environment, we would need to iterate over all enum values.

On the other hand, I found it useful to iterate over all supported enums for store/restore when I debugged the issue.

Reply:
@wujingyue perhaps we can move this discussion to a separate new PR?

This PR blocks changes that introduce support for matmul fusions with more than one output (epilogue modes with aux tensor from cuBLAS naming convention).

Reply:
I'm not following the would-need-to-iterate-over-all-enum-when-we-restore-environment part. Why wouldn't #1351 work?

Reply:
> @wujingyue perhaps we can move this discussion to a separate new PR?

SG. I dismissed my review and we can move this discussion to #1351.

Reply:
@wujingyue - now I see where I made a mistake, even though I modified the instance of options I got fixed that the API does not allow to modify stored option object. My bad and sorry for the noise.

I will incorporate your suggestion in this PR and proceed with its integration. Thanks!

Perhaps, to avoid misunderstandings, it would be better to add code snippets to the suggestion, to prevents mistakes like the one I made?